gridscale/resource_gridscale_k8s.go

```diff
@@ -583,48 +769,62 @@ func resourceGridscaleK8sRead(d *schema.ResourceData, meta interface{}) error {
 		return fmt.Errorf("%s error setting listen ports: %v", errorPrefix, err)
 	}
 
-	// Get node pool parameters
-	nodePoolList := make([]interface{}, 0)
-	// TODO: The API scheme will be CHANGED in the future. There will be multiple node pools.
-	nodePool := map[string]interface{}{
-		"name":         d.Get("node_pool.0.name"),
-		"node_count":   props.Parameters["k8s_worker_node_count"],
-		"cores":        props.Parameters["k8s_worker_node_cores"],
-		"memory":       props.Parameters["k8s_worker_node_ram"],
-		"storage":      props.Parameters["k8s_worker_node_storage"],
-		"storage_type": props.Parameters["k8s_worker_node_storage_type"],
-	}
+	// Iterate over node pools fetched from source to collect them
+	nodePoolsSetInterface, isNodePoolsSet := props.Parameters["pools"]
+	nodePools := make([]map[string]interface{}, 0)
 
-	// Set rocket storage if it is set
-	if _, isRocketStorageSet := props.Parameters["k8s_worker_node_rocket_storage"]; isRocketStorageSet {
-		nodePool["rocket_storage"] = props.Parameters["k8s_worker_node_rocket_storage"]
-	}
+	if isNodePoolsSet {
+		nodePoolsSet := nodePoolsSetInterface.([]map[string]interface{})
```

Comment:
@nvthongswansea I've followed your suggestion via interactive rebase.
Sure, I already considered that node pools is a list, where each node pool is supposed to be a string to interface mapping such as `map[string]interface{}`. This is why I initialized what we need to iterate over like `nodePoolsSet := nodePoolsSetInterface.([]map[string]interface{})`, which at least is a list of string to interface mapping. But when this caused an issue and we need to iterate over `nodePoolsSetInterface.([]interface{})` instead `nodePoolsSetInterface.([]map[string]interface{})`, then it's good to have it fixed now.
I think iterating over collection broken down via `nodePoolsSetInterface.([]map[string]interface{})` may break the flow in case if there are no node pools set, right?

Reply:
Yes. In this direction I was thinking. Thx.

Reply:
... Ah and I've recognized that at the functions called `resourceGridscaleK8sCreate`, `resourceGridscaleK8sUpdate` and `validateK8sParameters` I already had considered to break down the node pools interface into a collection of interfaces via `.([]interface{})`. So then I think at the function called `resourceGridscaleK8sRead` either it was a leftover or happened accidentally that i broke the interface down into a list of string to interface mapping via `.([]map[string]interface{})`.